### PR TITLE
fix: output umd build for main key

### PR DESCRIPTION
**Pull request by keithamus**
I did not notice in #15 but that PR ended up producing a build that only included ES modules. 

The `main` key in a package.json should not be used for files that produce es modules. We should instead produce a `umd` compatible bundle, and link `main` to that.

This PR changes the `build` step to produce both `usm` and `es` style modules, and changes the `main` key to point to the `umd` build.

### CODE REVIEW

**Comment by muan** [comment]
When we paired it was consciously decided that umd isn't useful to us anymore. Is satisfying what a `main` field expects the only reason to add it back? Is there anyway to work around that without creating a build file that we don't consume but need to maintain?

**Comment by keithamus** [comment]
We could omit the `main` field entirely, and rely only on the `module` key.

**Comment by dgraham** [comment]
Something to look forward to is Node's [experimental support](https://medium.com/@nodejs/announcing-a-new-experimental-modules-1be8d2d6c2ff) for a `"type": "module"` key in package.json. Once that stabilizes, we'll point `main` to the module file and won't need to build UMD files.

It looks like producing UMD output is fairly easy to do with command line overrides to `tsc`, so it would be easy to remove when we start using `type: module`.

File: test/karma.config.js [hunk 1]
```diff
@@ -25,13 +25,7 @@ function reply(request, response, next) {
...
   config.set({
     frameworks: ['mocha', 'chai'],
-    files: [
-      {
-        pattern: '../dist/index.js',
-        type: 'module'
-      },
-      'test.js'
-    ],
+    files: ['bootstrap.js', '../dist/umd/index.js', 'test.js'],
```
Comment (muan) [inline test/karma.config.js]: `bootstrap.js` seems like an over-complication in comparison to loading the native module. Can we keep the test using the module output here?

**Pull request merged**